Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java

```diff
@@ -174,12 +177,43 @@ public String search(HttpServletRequest req, ModelMap model) {
 	 *            A synonym string for which we need to find existing concepts
 	 * @return Returns array of concepts found for synonym name
 	 */
-	@RequestMapping(method = RequestMethod.GET, value = "conceptWrapperAddSynonymView")
-	public @ResponseBody
-	ConceptEntry[] getSynonyms(@RequestParam("synonymname") String synonymname) {
-		ConceptEntry[] entries = conceptManager
-				.getConceptListEntriesForWord(synonymname.trim());
-		return entries;
-	}
+    @RequestMapping(method = RequestMethod.GET, value = "conceptWrapperAddSynonymView")
+    public ResponseEntity<String> getSynonyms(@RequestParam("synonymname") String synonymname) {
+        ConceptEntry[] entries = conceptManager.getConceptListEntriesForWord(synonymname.trim());
+        return new ResponseEntity<String>(buildJSON(Arrays.asList(entries)), HttpStatus.OK);
+    }
+
+    private String buildJSON(List<ConceptEntry> synonyms) {
+
+        int i = 0;
+        StringBuffer jsonStringBuilder = new StringBuffer("{");
+        jsonStringBuilder.append("\"Total\"");
+        jsonStringBuilder.append(":");
+        jsonStringBuilder.append(synonyms.size());
+        jsonStringBuilder.append(",");
+        jsonStringBuilder.append("\"synonyms\":");
+        jsonStringBuilder.append("[");
+        for (ConceptEntry syn : synonyms) {
+            // Appending for next element in JSON
+            if (i != 0) {
+                jsonStringBuilder.append(",");
+            }
+
+            jsonStringBuilder.append("{");
+            jsonStringBuilder.append("\"id\":\"" + syn.getId() + "\"");
+            jsonStringBuilder.append(",");
+            jsonStringBuilder.append("\"word\":\"" + syn.getWord() + "\"");
+            jsonStringBuilder.append(",");
+            jsonStringBuilder.append("\"description\":\"" + syn.getDescription().replaceAll("\"", "'") + "\"");

```

Comment:
you probably want to escape the content of the JSON here already and not replace all quotes with single quotes.